### PR TITLE
feat(retry tx): Initialize retry-tx command PE-1246

### PR DESCRIPTION
**Pull request by fedellen**
This PR adds the CLI side functionality for the `retry-tx` command.

~~Currently in draft, still need to add the README component~~

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -951,6 +952,34 @@ Example output:
...
+
+#### Information Required to Retry a Transaction
+
+In order to re-seed the chunks to an unbundled ArFS data transaction, a user must have the data transaction ID, the original file data, and a destination folder ID or a valid file ID for the file.
+
+When an upload fails, generally this transaction ID information would be lost. But a user can quickly recover this transaction ID with the `ardrive last-tx -w /path/to/wallet` command.
+
+#### Retrying with ArDrive CLI
+
+With the data transaction ID and the file data is recovered a file simply use the `retry-tx` command:
```
Comment (karlprieb) [inline README.md:975]: I'm having a hard time to understand this phrase. I feel it doesn't make sense, maybe because English is not my main language.

**Review by karlprieb — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -951,6 +952,22 @@ Example output:
...
+
+Arweave data upload transactions are split into two phases: transaction posting and chunks uploading. Once the transaction post phase has been completed, you've effectively "paid" the network for storage of the data chunks that you'll send in the next stage.
+
+If your system encounters an error while posting the transaction, you can retry posting the transaction for as long as your tx_anchor is valid ([learn more about tx_anchors here][tx_anchors]). You may retry and/or resume posting chunks at any time after your transaction has posted. The ArDrive CLI allows you to take advantage of this Arweave protocol capability.
+
+Using the CLI, when the transaction post has succeeded but the chunk upload fails due to the process crashing the data transaction's ID could be lost. But a user has a few options to recover this ID. If it is their last transaction, they can quickly recover this transaction ID with the `ardrive last-tx -w /path/to/wallet` command. Another option us to use an Arweave gateway GQL endpoint to search for transactions that belong to their wallet. Or they could check out their latest transactions with a third party tool like [ViewBlock][viewblock].
```
Comment (arielmelendez) [inline README.md:961]:
```suggestion
Using the CLI, when the transaction post has succeeded but the chunk upload step fails, the data transaction's ID could be lost. There are a few options to recover this ID. If the failed transaction is the most recent one sent from a wallet, the transaction ID can be quickly recovered with the `ardrive last-tx -w /path/to/wallet` command. Other options for finding the partially uploaded transaction's ID include: 
- Using an Arweave gateway GQL http endpoint to search for transactions that belong to the wallet. See this [Arweave GQL Guide][https://gql-guide.vercel.app/#owners] for more info.
- Browse the recent  transactions associated with the wallet via a block explorer tool like [ViewBlock][viewblock].
```

File: README.md [hunk 3]
```diff
@@ -951,6 +952,22 @@ Example output:
...
+
+In order to re-seed the chunks to an unbundled ArFS data transaction, a user must have the data transaction ID, the original file data, and a destination folder ID or a valid file ID for the file. With this information simply use the `retry-tx` command:
```
Comment (arielmelendez) [inline README.md:963]:
```suggestion
In order to re-seed the chunks for an unbundled ArFS data transaction, a user must have the data transaction ID, the original file data, and either a destination folder ID or a valid file ID for the file. Supply that information to the `retry-tx` command like so:
```

File: README.md [hunk 4]
```diff
@@ -951,6 +952,22 @@ Example output:
...
+
+```shell
+ardrive retry-tx --transaction-id { Data Transaction ID } --parent-folder-id { Destination Folder ID }  --local-path /path/to/file  --wallet-file /path/to/wallet
+```
+
+**Note: Retry feature is currently only available for PUBLIC unbundled file transactions. It is also perfectly safe to mistakenly re-seed the chunks of a healthy transaction, the transaction will remain stable and the wallet will not be re-charged.**
```
Comment (arielmelendez) [inline README.md:969]:
```suggestion
**Note: Retry feature is currently only available for PUBLIC unbundled file transactions. It is also perfectly safe to mistakenly re-seed the chunks of a healthy transaction, the transaction will remain stable and the wallet balance will not be affected.**
```

File: src/commands/retry_tx.ts [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+		LocalPathParameter,
+		{ name: ParentFolderIdParameter, required: false },
+		{ name: FileIdParameter, required: false },
+		DestinationFileNameParameter,
+		BoostParameter,
+		DryRunParameter,
+		...ConflictResolutionParams,
+		CustomContentTypeParameter,
+		GatewayParameter,
+		{ name: TransactionIdParameter, description: 'The data transaction id in which to retry' },
```
Comment (arielmelendez) [inline src/commands/retry_tx.ts:33]:
```suggestion
		{ name: TransactionIdParameter, description: 'The transaction ID of the data transaction to retry' },
```

File: src/commands/retry_tx.ts [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+
+		const arDrive = cliArDriveFactory({
+			wallet,
+			feeMultiple: parameters.getOptionalBoostSetting(),
+			dryRun: parameters.isDryRun(),
+			arweave
+		});
+
+		const destinationFolderId = parameters.getParameterValue(ParentFolderIdParameter, EID);
+		const fileId = parameters.getParameterValue(FileIdParameter, EID);
```
Comment (arielmelendez) [inline src/commands/retry_tx.ts:61]: We should assert here to the user that one or the other is provided.

**Review by arielmelendez — CHANGES_REQUESTED** [review]
Looks really good! Please review my suggested updates.

File: package.json [hunk 7]
```diff
@@ -8,7 +8,7 @@
 	},
 	"types": "./lib/index.d.ts",
 	"dependencies": {
-		"ardrive-core-js": "1.13.0",
+		"ardrive-core-js": "1.13.0-alpha-1",
```
Comment (matibat) [inline package.json:11]: Reminder to set the correct version after we release core

**Review by matibat — APPROVED** [review]
It looks great! Glad we'll have this new command 🥳

**Review by arielmelendez — DISMISSED** [review]

**Review by matibat — APPROVED** [review]
🚀

**Pull request merged**